modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/AddToCartButton.js

```diff
@@ -58,60 +128,29 @@ function AddToCartButton({
 			disabled={disabled}
 			displayType="primary"
 			monospaced={settings.iconOnly && settings.inline}
-			onClick={(event) => {
-				if (cartAtomState.updating) {
-					return;
-				}
+			onClick={async (event) => {
+				const CartResource = ServiceProvider.DeliveryCartAPI('v1');
+				const order = await CartResource.getCartsByAccountIdAndChannelId(
```

Comment:
this looks like the problem with the guest order, why did you add this?

Reply:
this is anyway returning an array of order not just one, how do you select the right one?

Reply:
I added this because we need to get the order count. If there is no order, I'll create a new order with the selected order type.